### ISSUE TITLE
Codewalking to infer types

### ISSUE THREAD

**Issue by digikar99** [robert-strandh/Concrete-Syntax-Tree#37]
Would this library be suitable for inferring variable types, for example, can I use it to determine the type of `a` in:

```lisp
(let ((a 5)) a) ; I'd expect (EQL 5)
;;; OR
(let ((a 5)) ; I'd expect (OR (EQL 5) (EQL 6))
  (setq a 6)
  a)
```

If so, is there any example code about how I can achieve this?

**Comment by robert-strandh** [comment]
You would probably need something like Trucler for that. 
It is a library for managing lexical compile-time environments.

**Comment by digikar99** [comment]
I see, thank you!
But so, while trucler provides the environment protocol, a code-walker that uses it is still yet to be developed, is that the case?

**Comment by robert-strandh** [comment]
Right.  I am unaware of any general-purpose code walker having been developed
using Trucler.  I suspect it has to do with the incomplete support that Trucler
provides for the widely used Common Lisp implementations.
